test-suite/src/test/require-jdk8/org/scalajs/testsuite/javalib/time/LocalDateTest.scala

```diff
@@ -0,0 +1,790 @@
+        }
+        for (n <- Seq(Long.MinValue, 0L, 6L, Long.MaxValue)) {
+          expectThrows[DateTimeException](d.`with`(ALIGNED_WEEK_OF_MONTH, n))
+        }
+        for (n <- Seq(Long.MinValue, 0L, 54L, Long.MaxValue)) {
+          expectThrows[DateTimeException](d.`with`(ALIGNED_WEEK_OF_YEAR, n))
+        }
+        for (n <- Seq(Long.MinValue, 0L, 13L, Long.MaxValue)) {
+          expectThrows[DateTimeException](d.`with`(MONTH_OF_YEAR, n))
+        }
+        for (n <- Seq(Long.MinValue, -11999999989L, 12000000000L, Long.MaxValue)) {
+          expectThrows[DateTimeException](d.`with`(PROLEPTIC_MONTH, n))
+        }
+        for (n <- Seq(Long.MinValue, 0L, 1000000001L, Long.MaxValue)) {
+          expectThrows[DateTimeException](d.`with`(YEAR_OF_ERA, n))
+        }
+        for (n <- Seq(Long.MinValue, -1000000000L, 1000000000L, Long.MaxValue)) {
+          expectThrows[DateTimeException](d.`with`(YEAR, n))
+        }
+        for (n <- Seq(Long.MinValue, -1L, 2L, Long.MaxValue)) {
+          expectThrows[DateTimeException](d.`with`(ERA, n))
+        }
+      }
+    }
+
+    it("should respond to `withYear`") {
+      testTemporal(MIN.withYear(-999999999))(MIN)
+      testTemporal(MIN.withYear(999999999))(of(999999999, 1, 1))
+      testTemporal(MAX.withYear(-999999999))(of(-999999999, 12, 31))
+      testTemporal(MAX.withYear(999999999))(MAX)
+
+      for {
+        d <- samples
+        n <- Seq(Int.MinValue, -1000000000, 1000000000, Int.MaxValue)
+      } {
+        expectThrows[DateTimeException](d.withYear(n))
+      }
+    }
+
+    it("should respond to `withMonth`") {
+      testTemporal(MAX.withMonth(2))(of(999999999, 2, 28))
+      testTemporal(MAX.withMonth(11))(of(999999999, 11, 30))
+      testTemporal(someDate.withMonth(1))(of(2011, 1, 28))
+      testTemporal(leapDate.withMonth(2))(leapDate)
+
+      for {
+        d <- samples
+        n <- Seq(Int.MinValue, 0, 13, Int.MaxValue)
+      } {
+        expectThrows[DateTimeException](d.withMonth(n))
+      }
+    }
+
+    it("should respond to `withDayOfMonth`") {
+      testTemporal(someDate.withDayOfMonth(1))(of(2011, 2, 1))
+      testTemporal(leapDate.withDayOfMonth(28))(of(2012, 2, 28))
+
+      expectThrows[DateTimeException](someDate.withDayOfMonth(29))
+      expectThrows[DateTimeException](leapDate.withDayOfMonth(30))
+      expectThrows[DateTimeException](of(0, 4, 30).withDayOfMonth(31))
+      for {
+        d <- samples
+        n <- Seq(Int.MinValue, 0, 32, Int.MaxValue)
+      } {
+        expectThrows[DateTimeException](d.withDayOfMonth(n))
+      }
+    }
+
+    it("should respond to `withDayOfYear`") {
+      testTemporal(someDate.withDayOfYear(1))(of(2011, 1, 1))
+      testTemporal(someDate.withDayOfYear(365))(of(2011, 12, 31))
+      testTemporal(leapDate.withDayOfYear(366))(of(2012, 12, 31))
+
+      expectThrows[DateTimeException](someDate.withDayOfYear(366))
+      for {
+        d <- samples
+        n <- Seq(Int.MinValue, 0, 367, Int.MaxValue)
```

Comment:
Define the `Seq` before the the `for` to avoid recreating it on each outer loop cycle.